src/zenml/integrations/huggingface/deployers/huggingface_deployer.py

```diff
@@ -295,10 +300,22 @@ def do_provision_deployment(
         )
 
         try:
-            # Create Space if it doesn't exist
+            # Create Space if it doesn't exist, or update visibility if needed
             try:
-                api.space_info(space_id)
+                space_info = api.space_info(space_id)
                 logger.info(f"Updating existing Space: {space_id}")
+
+                # Update visibility if changed
+                if space_info.private != settings.private:
+                    logger.info(
+                        f"Updating Space visibility: "
+                        f"{'private' if settings.private else 'public'}"
+                    )
+                    api.update_repo_visibility(
```

Comment:
From the HF docs: "update_repo_visibility method has been removed. Please use update_repo_settings instead." (https://huggingface.co/docs/huggingface_hub/v1.0.0.rc7/en/concepts/migration#other-deprecated-features)